### ISSUE TITLE
Investigation with thousands of entities will not open

### ISSUE THREAD

**Issue by Jermain-N** [LuatixHQ/opencti#7219]
## Description

I have created a grouping that contains 114 reports. The reports, all together, contain 4819 indicators. WHen I expand the reports in the investigation to dhow all the indicators, it takes a couple minutes to refresh the page and add the indicators to the investigation.

## Environment

1. OS (where OpenCTI server runs): hosted in Filigran Cloud
2. OpenCTI version: OpenCTI 6.1.8
3. OpenCTI client: Frontend
4. Other environment details: navigating OpenCTI with Chrome v125

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create a grouping and add to it a large volume of reports (over 110) that contain, all together, a VERY large volme of indicators (over 4800 indicators).
2. Create a new investigation.
3. Add the grouping to the investigation.
4. Select and expand the grouping.
5. Select and expand the reports (just expand indicators, not the other elements).
6. Navigate away from the investigation (e.g. go to a dashboard) OR just refresh the investigation page (press F5).
7. Error appears on screen.

## Expected Output

Expecting the investigation entities to reappear. There is nothing in the documentation detailing an explicit limit on how many entities can be displayed in an investigation.

## Actual Output

Error message (see screenshot)

## Additional information

I suspect it's the too large volume of entities being displayed that is causing the error.

## Screenshots (optional)

![image](https://github.com/OpenCTI-Platform/opencti/assets/155680154/70073e20-4db2-486d-bdde-a25c45fdcdd1)

![image](https://github.com/OpenCTI-Platform/opencti/assets/155680154/6011625e-ecc7-4bfb-a03f-febeccaa4f27)

![image](https://github.com/OpenCTI-Platform/opencti/assets/155680154/fe3db971-cb4b-48ac-a9b8-319aebe5aa64)

![image](https://github.com/OpenCTI-Platform/opencti/assets/155680154/3d24852a-f526-4cef-8c59-02ad4c23a653)

**Comment by nino-filigran** [comment]
My Investigation simply crashed when I tried to expand (around 6000 IOCs). I'm removing the triage.

**Comment by Jermain-N** [comment]
@nino-filigran Hi Nino, do we have an exact number on how many IOCs can be displayed in the investigation feature?

**Comment by nino-filigran** [comment]
I don't have the answer @Jermain-N but can try to find it. Overall this would not be about IOCs only but about any type of entities.

**Comment by Jermain-N** [comment]
@nino-filigran Yes that was an "abus de langage" on my behalf, the maximum number of any entities in a knowledge graph is the number I'm looking for.

**Comment by nino-filigran** [comment]
Given we have done something in 6.6 to be able to load big grpah by paginating the loading, I'm closing this bug for now.

**Comment by labo-flg** [comment]
Confirmed: it should now work correctly in 6.6.0, after the major refactoring we've done in the graph code base. Loading has been improved with pagination and progress bar for user feedback.
